File: hailiao_flutter/lib/models/friend_request_dto.dart
import 'user_dto.dart';

class FriendRequestDTO {
  final int? id;
  final int? fromUserId;
  final int? toUserId;
  final String? remark;
  final String? message;
  final int? status;
  final String? handledAt;
  final String? createdAt;
  final UserDTO? fromUserInfo;
  final UserDTO? toUserInfo;

  FriendRequestDTO({
    this.id,
    this.fromUserId,
    this.toUserId,
    this.remark,
    this.message,
    this.status,
    this.handledAt,
    this.createdAt,
    this.fromUserInfo,
    this.toUserInfo,
  });

  factory FriendRequestDTO.fromJson(Map<String, dynamic> json) {
    return FriendRequestDTO(
      id: (json['id'] as num?)?.toInt(),
      fromUserId: (json['fromUserId'] as num?)?.toInt(),
      toUserId: (json['toUserId'] as num?)?.toInt(),
      remark: json['remark'] as String?,
      message: json['message'] as String?,
      status: (json['status'] as num?)?.toInt(),
      handledAt: json['handledAt'] as String?,
      createdAt: json['createdAt'] as String?,
      fromUserInfo: json['fromUserInfo'] is Map<String, dynamic>
          ? UserDTO.fromJson(json['fromUserInfo'] as Map<String, dynamic>)
          : null,
      toUserInfo: json['toUserInfo'] is Map<String, dynamic>
          ? UserDTO.fromJson(json['toUserInfo'] as Map<String, dynamic>)
          : null,
    );
  }
}

File: hailiao-api/src/main/java/com/hailiao/api/dto/FriendRequestDTO.java
package com.hailiao.api.dto;

import java.util.Date;

public class FriendRequestDTO {
    private Long id;
    private Long fromUserId;
    private Long toUserId;
    private String remark;
    private String message;
    private Integer status;
    private Date handledAt;
    private Date createdAt;
    private UserDTO fromUserInfo;
    private UserDTO toUserInfo;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getFromUserId() { return fromUserId; }
    public void setFromUserId(Long fromUserId) { this.fromUserId = fromUserId; }
    public Long getToUserId() { return toUserId; }
    public void setToUserId(Long toUserId) { this.toUserId = toUserId; }
    public String getRemark() { return remark; }
    public void setRemark(String remark) { this.remark = remark; }
    public String getMessage() { return message; }
    public void setMessage(String message) { this.message = message; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public Date getHandledAt() { return handledAt; }
    public void setHandledAt(Date handledAt) { this.handledAt = handledAt; }
    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }
    public UserDTO getFromUserInfo() { return fromUserInfo; }
    public void setFromUserInfo(UserDTO fromUserInfo) { this.fromUserInfo = fromUserInfo; }
    public UserDTO getToUserInfo() { return toUserInfo; }
    public void setToUserInfo(UserDTO toUserInfo) { this.toUserInfo = toUserInfo; }
}
